### ISSUE TITLE
Collective page redirects

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2283]
## User story 

- As a collective core contributor, I get my previous URL redirected to the new one when I change the slug of my collective.

- As an Open Collective user, I don't want to loose the SEO of my profile when we change the slug.

- As an Open Collective admin, I can manually setup multiple redirects (for example I can decide that `/opensource`, `/open-source` should bother redirect to `/osc`)

## Best solution for this problem

- **API**
  - [ ] Create a `CollectiveSlugRedirects` table with:
    - `redirected` (string)
    - `redirectedTo` (string)
  - [ ] Ensure we return the same error type (`CollectiveNotFound`) when no collective is found for a slug
  - [ ] Whenever get a `CollectiveNotFound` exception in the GraphQL API, check the redirect table and transform the error in a `CollectiveRedirected` including the new slug if we find one.
- **Frontend**
  - [ ] In the `Error` component, whenever we get a `CollectiveRedirected` error from the API, show a loading spinner with `Redirecting...` and redirect to the same page we're in just replacing the collective slug.
  - [ ] Return the proper 302 HTTP code to ensure search engines update their references

## MVP
We can manually setup the redirections at first when requested (as in https://opencollective.freshdesk.com/a/tickets/951).

## Metrics 
- How many redirects we catch
- SEO improvements

## Documentation 
We should add a line in the documentation to mention that collectives can contact us to setup a custom redirect.

**Comment by RichardLitt** [comment]
I would like this to be solved.

**Comment by debryc** [comment]
I completely agree! We're making a pretty significant name change and with a lack of URL redirect, all of our previous links break.

**Comment by debryc** [comment]
I just realized: does changing the slug also break _all_ previous links to individual transactions as well? If so, that would be disastrous for us. As a fiscal host, we link to transactions within our QuickBooks Online system and any audit would require being able to connect transactions in Quickbooks with the corresponding transactions in Open Collective.

**Comment by Betree** [comment]
> does changing the slug also break all previous links to individual transactions as well?

@debryc It does. If you rename `my-collective` to `my-collective-new`, the URL for https://opencollective.com/my-collective/orders/182079 will be renamed to https://opencollective.com/my-collective-new/orders/182079. 

support@opencollective.com can eventually help to redirect these URLs if you mistakingly end up in such situation, but we don't officially support this feature right now, so any change in the `slug` should be considered as breaking all links that have been shared for the edited account.

**Comment by debryc** [comment]
@Betree Whew. I'm glad I checked about the URL!

We have very strong reasons for wanting to rename our collectives and given there's no support for the URL redirect at the moment, I think the sooner we change the slugs the better so there are fewer transactions to go through. Would you all be so kind as to help us change the collective slug and associated URLs safely?

**Comment by Betree** [comment]
@debryc Sorry, as mentioned, the feature is not currently supported. I'd recommend not changing the slug if you've shared them in important places or if critical parts of your workflow depend on it.

**Comment by Betree** [comment]
I got a case today of a collective that changed the URL without realizing it would break all the URLs they've shared across the web. Until we implement this issue, we should at least add a warning before saving to make sure people realize the impact that changing the URL will have.

**Comment by SudharakaP** [comment]
This come up from time to time, the most recent one being https://opencollective.freshdesk.com/a/tickets/369380
